src/transport_performance/osm/validate_osm.py

```diff
@@ -0,0 +1,682 @@
+
+    """
+
+    def __init__(self, osm_pth) -> None:
+        super().__init__()
+        _is_expected_filetype(osm_pth, "osm_pth", exp_ext=".pbf")
+        self.apply_file(osm_pth, locations=True)
+        self.found_locs = dict()
+
+    def check_locs_for_ids(self, ids: list, feature_type: str) -> dict:
+        """Return coordinates for provided list of feature IDs.
+
+        Parameters
+        ----------
+        ids : list
+            A list of OSM feature IDs to check. IDs must be integer.
+        feature_type : str
+            The type of feature to which the IDS belong. Valid options are
+            "node" or "way".
+
+        Returns
+        -------
+        found_locs: dict
+            ID: dict of tags, containing ID : location.
+
+        """
+        self.found_locs = _filter_target_dict_with_list(
+            targets={"node": self.node_locs, "way": self.way_node_locs},
+            _list=ids,
+            search_key=feature_type,
+            accepted_keys=["node", "way"],
+        )
+        return self.found_locs
+
+    def plot_ids(
+        self,
+        ids: list,
+        feature_type: str,
+        crs: Union[str, int] = "epsg:4326",
+    ) -> folium.Map:
+        """Plot coordinates for nodes or node members of a way.
+
+        Provided with a list of node or way IDs, converts the coordinate data
+        from dictionary to GeoDataFrame and uses the basic gdf.explore() method
+        to visualise the features on a basemap.
+
+        Parameters
+        ----------
+        ids : list
+            A list of Node or Way IDs.
+        feature_type : str
+            Whether the type of OSM feature to plot is node or way.
+        crs : Union[str, int], optional
+            The projection of the spatial features, by default "epsg:4326"
+
+        Returns
+        -------
+        folium.Map
+            A plot of the coordinate data for each identifiable node.
+
+        Raises
+        ------
+        NotImplementedError
+            Relation location data could be extracted with a bit more munging.
+            Please raise a feature request if you feel this is significant.
+        ValueError
+            `feature_type` is not one of "node", "way", "relation" or "area".
+        TypeError
+            `ids` is not of type list.
+            `feature_type` is not of type str.
+
+        """
+        _type_defence(ids, "ids", list)
+        _type_defence(feature_type, "feature_type", str)
+        _type_defence(crs, "crs", (str, int))
+        feature_type = feature_type.lower().strip()
+        ACCEPT_FEATS = ["node", "way", "relation", "area"]
+        _check_item_in_iter(
+            item=feature_type, iterable=ACCEPT_FEATS, param_nm="feature_type"
+        )
+        if feature_type != "node" and feature_type != "way":
+            raise NotImplementedError(
+                "Relation or area plotting not implemented at this time."
+            )
```

Comment:
Would this implementation be more suitable. Functionality is the same however this seems more dynamic for adding other features.  

```python
if feature_type not in ["node", "way"]:
    raise NotImplementedError(
       "Relation or area plotting not implemented at this time."
    )
```